### DISCUSSION TITLE
Upgrade Jetty version without upgrading Mirth

### DISCUSSION THREAD

**Discussion by mayurbhatkhande** [nextgenhealthcare/connect#4520] [Q&A] [answered]
Hello All,
We are using Mirth version 2.2.1. 
It uses Java 1.6 and Jetty 7.5.4. 
We need to upgrade jetty version to 9.4 which works on Java 1.8.
Does mirth 2.2.1 supports Jetty 9.4 or I must upgrade Mirth itself ?

Thanks.

**Comment by jonbartels** [comment, marked as answer]
You really should upgrade MC itself. 2.2.1 is a very old release, Java 1.6 is also very old. It does not seem likely that you'd be able to upgrade only Jetty and not have issues. Jetty has changed a lot since 7.x to 9.x. The libraries that Jetty 9.x depends on would also likely conflict with other parts of MC.

Most users only ask about Jetty versions when a security scan detects an old version of Jetty. This usually happens from looking at HTTP headers. If you have a security scan that detects Jetty then it seems very likely that a deeper or different scan will also find outdated versions of Java installed. 

Upgrading MC and running a recent Java release is a much safer posture.

The open source community might be able to help guide your upgrade. 
How many channels are installed on the MC instances that you need to upgrade? 
Do you have a good TEST environment with traffic similar to your PROD environment?

**Reply by mayurbhatkhande** [reply]
Thanks jonbartels for your reply!
For now, I don't have any questions regarding upgrade.

Regards,
